src/main/java/pl/edu/pk/iti/copperAnt/simulation/Clock.java

```diff
@@ -54,6 +57,11 @@ public int getNumberOfWaitingEvent() {
 	public void run() {
 		while (!finishCondition.isSatisfied(this)) {
 			tick();
+			try {
+		        Thread.sleep(50);
```

Comment:
Twoja sugestia naświetliła mi w czym jest problem, ale dodanie tutaj sleep nie jest najlepsze bo wątek i tak śpi w  metodzie tick(). Problem wynika z tego, że w metodzie sleep wątek śpi czas proporcjonalny do czasu zdarzeń, o one są źle ustalone. Proponuje w ramach quick solution nie dodawania tutaj sleepa, ale zmiane domyślnego timeScale zegara. Jeśli nie masz nic przeciwko zmerguje twój branch to siebie i wprowadzę tą zmiąne już u siebie.